secure_message/common/utilities.py

```diff
@@ -70,68 +70,75 @@ def process_paginated_list(paginated_list, host_url, user, message_args, endpoin
+def get_to_details(messages):
+    """Adds a @msg_to key every message in a list of messages.
+    Every msg_to uuid is resolved to include details of the user.
 
-    details = []
+    If the call for the internal user id fails, an exception will be thrown.
+    If the external user id cannot be found in the list that we got from the party service.  There
+    won't be a @msg_to value returned in the payload.  The API documentation notes that these elements
+    aren't guaranteed to be provided so we're not breaking the contract by doing this.
+    """
 
-    for ru in rus:
-
-        detail = party.get_business_details(ru)
-
-        if detail is not None:
-            details.append(detail)
+    external_user_details = party.get_users_details(get_external_user_uuid_list(messages))
+    for message in messages:
+        if not message["from_internal"]:
+            message.update({"@msg_to": [internal_user_service.get_user_details(message["msg_to"][0])]})
         else:
-            logger.info('No details found for RU ID', ru=ru)
+            for detail in external_user_details:
+                if detail['id'] == message['msg_to'][0]:
+                    message.update({'@msg_to': [detail]})
+                    break
 
-    return details
-
-
-def add_to_and_from_details(messages):
-    """Adds user details for sender and recipient"""
-    for message in messages:
-        message.update({'@msg_from': _get_from_details(message), '@msg_to': _get_to_details(message)})
     return messages
 
 
-def _get_from_details(message):
-    """looks up the details for the from users"""
-    logger.info("Getting user from details", is_internal=message['from_internal'], user_uuid=message['msg_from'])
-    if message['from_internal']:
-        from_details = internal_user_service.get_user_details(message['msg_from'])
-    else:
-        from_details = party.get_user_details(message['msg_from'])
-    return from_details
+def get_from_details(messages):
+    """Adds a @msg_from key every message in a list of messages.
+    Every msg_to uuid is resolved to include details of the user.
 
+    If the call for the internal user id fails, an exception will be thrown.
+    If the external user id cannot be found in the list that we got from the party service.  There
+    won't be a @msg_from value returned in the payload.  The API documentation notes that these elements
+    aren't guaranteed to be provided so we're not breaking the contract by doing this.
+    """
+    external_user_details = party.get_users_details(get_external_user_uuid_list(messages))
+    for message in messages:
+        if message["from_internal"]:
+            message.update({"@msg_from": internal_user_service.get_user_details(message["msg_from"])})
+        else:
+            for detail in external_user_details:
+                if detail['id'] == message['msg_from']:
+                    message.update({'@msg_from': detail})
```

Comment:
If we took the data returned from the party service and structured it like:
```
details = {
    'party_uuid1': {
		...party_detail...
    },
    'party_uuid2': {
		...party_detail...
    },
    ...
}
```
we wouldn't need to loop over `external_user_details` for every message, we could do:
```
if details.get(msg_from):
	message.update({'@msg_from': details.get(msg_from)})
```

Reply:
Updated with cleaner way